### PR TITLE
Enable json output from solana feature status

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
As the managed feature set gets bigger for a cluster, it will be useful to be able to process status information from the Cli programmatically.

#### Summary of Changes
Add OutputFormat handling for `cli::features::process_status`

Examples:

```
$ solana feature status --url http://devnet.solana.com --output json
{
  "features": [
    {
      "id": "3h1BQWPDS5veRsq6mDBWruEpgPxRJkfwGexg5iiQ9mYg",
      "description": "consistent recentblockhashes sysvar",
      "status": "active",
      "sinceSlot": 5414912
    },
    {
      "id": "DFBnrgThdzH4W6wZ12uGPoWcMnvfZj11EHnxHcVxLPhD",
      "description": "bpf_loader2 program",
      "status": "active",
      "sinceSlot": 5414912
    },
    {
      "id": "E3PHP7w8kB7np3CTQ1qQ2tW3KCtjRSXBQgW9vM2mWv2Y",
      "description": "secp256k1 program",
      "status": "active",
      "sinceSlot": 5414912
    },
    {
      "id": "E5JiFDQCwyC6QfT9REFyMpfK2mHcmv1GUDySU1Ue7TYv",
      "description": "spl-token multisig fix",
      "status": "active",
      "sinceSlot": 5414912
    },
    {
      "id": "EnvhHCLvg55P7PDtbvR1NwuTuAeodqpusV3MR5QEK8gs",
      "description": "instructions sysvar",
      "status": "active",
      "sinceSlot": 5414912
    },
    {
      "id": "GaBtBJvmS4Arjj5W1NmFcyvPjsHN38UGYDq2MDwbs9Qu",
      "description": "pico-inflation",
      "status": "active",
      "sinceSlot": 5414912
    }
  ],
  "featureActivationAllowed": true
}
```

```
$ solana feature status 3h1BQWPDS5veRsq6mDBWruEpgPxRJkfwGexg5iiQ9mYg --url http://testnet.solana.com --output json
{
  "features": [
    {
      "id": "3h1BQWPDS5veRsq6mDBWruEpgPxRJkfwGexg5iiQ9mYg",
      "description": "consistent recentblockhashes sysvar",
      "status": "inactive"
    }
  ],
  "featureActivationAllowed": false
}
```

### CODE REVIEW

File: cli-output/src/cli_output.rs [hunk 1]
```diff
@@ -1352,6 +1352,71 @@ pub fn parse_sign_only_reply_string(reply: &str) -> SignOnly {
     }
 }
 
+#[derive(Serialize, Deserialize)]
```
Comment (mvines) [inline cli-output/src/cli_output.rs:1355]: Could these things live in `cli/src/feature.rs`?    I really like how `cli/src/feature.rs` exposes a minimal surface area to the rest of the cli code

Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs:1355]: Yes, I agree! I had considered busting up cli_output as a follow-up sweeping change, but I'll go ahead and move these now.

Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs]: Updated

Comment (t-nelson) [reply cli-output/src/cli_output.rs]: > I had considered busting up cli_output

+1 Plenty of stuff in here that's CLI-specific and will never be useful elsewhere

**Review by mvines — APPROVED** [review]
Fully contained within `feature.rs`! :bowtie:

**Pull request merged**